recoverytool/src/main.rs

```diff
@@ -0,0 +1,270 @@
+    /// Path to database
+    #[arg(long = "db")]
+    db: PathBuf,
+    /// Directory containing server config files
+    #[arg(long = "cfg")]
+    config: PathBuf,
+    /// The password that encrypts the configs, will prompt if not passed in
+    #[arg(long = "password", env = "FM_PASSWORD")]
+    password: Option<String>,
+}
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| EnvFilter::new("error,mint_client=info,fedimint_cli=info")),
+        )
+        .with_writer(std::io::stderr)
+        .init();
+
+    let opts: RecoveryTool = RecoveryTool::parse();
+
+    let salt_path = opts.config.join(SALT_FILE);
+    let key = get_key(opts.password, salt_path).expect("Failure deriving encryption key");
+    let cfg = read_server_configs(&key, opts.config).expect("Could not read config file");
+    let wallet_cfg: WalletConfig = cfg
+        .get_module_config_typed(LEGACY_HARDCODED_INSTANCE_ID_WALLET)
+        .expect("Malformed wallet config");
+
+    let db = Database::new(
+        RocksDb::open(opts.db).expect("Error opening DB"),
+        Default::default(),
+    );
+
+    let db = if opts.legacy {
+        db
+    } else {
+        db.new_isolated(LEGACY_HARDCODED_INSTANCE_ID_WALLET)
+    };
+
+    let ctx = secp256k1::Secp256k1::new();
+    let base_descriptor = wallet_cfg.consensus.peg_in_descriptor;
+    let base_key = wallet_cfg.private.peg_in_key;
+
+    let utxos: Vec<ImportableWallet> = db
+        .begin_transaction()
+        .await
+        .find_by_prefix(&UTXOPrefixKey)
+        .await
+        .map(|res| {
+            let (UTXOKey(outpoint), SpendableUTXO { tweak, amount }) = res.expect("DB error");
+            let secret_key = base_key.tweak(&tweak, &ctx);
+            let pub_key =
+                CompressedPublicKey::new(secp256k1::PublicKey::from_secret_key_global(&secret_key));
+            let descriptor = base_descriptor
+                .tweak(&tweak, &ctx)
+                .translate_pk(&mut SecretKeyInjector {
+                    secret: bitcoin::util::key::PrivateKey {
+                        compressed: true,
+                        network: wallet_cfg.consensus.network,
+                        inner: secret_key,
+                    },
+                    public: pub_key,
+                })
+                .expect("can't fail");
+
+            ImportableWallet {
+                outpoint,
+                descriptor,
+                amount_sat: amount,
+            }
+        })
+        .collect()
+        .await;
+
+    serde_json::to_writer(std::io::stdout().lock(), &utxos).expect("Could not encode to stdout")
+}
+
+/// A UTXO with its Bitcoin Core importable descriptor
+#[derive(Debug, Serialize)]
+struct ImportableWallet {
+    outpoint: OutPoint,
+    descriptor: Descriptor<Key>,
+    #[serde(with = "bitcoin::util::amount::serde::as_sat")]
+    amount_sat: Amount,
```

Comment:
nite: I think it's more readable to use `bitcoin::Amount`

Reply:
done